callBackPractice/callBackPractice.js

```diff
@@ -20,52 +20,79 @@
 
 const foods = ['pineapple', 'mango', 'ribeye', 'curry', 'tacos', 'ribeye', 'mango'];
 
+const firstItem = (arr, cb) => {
+  cb(arr[0]);
+};
+
 firstItem(foods, (firstItem) => {
   console.log(`The first item is ${firstItem}.`);
 });
 
 // Write a function called getLength that passes the length of the array into the callback
+const getLength = (arr, cb) => {
+  cb(arr.length);
+};
 
 getLength(foods, (length) => {
   console.log(`The length of the array is ${length}.`);
 });
 
 // Write a function called last which passes the last item of the array into the callback
+//not working
+const lastItem = (arr, cb) => {
+  cb(arr[arr.length-1]);
+};
 
 last(foods, (lastItem) => {
   console.log(`The last item in the array is ${lastItem}.`);
 });
 
 // Write a function called sumNums that adds two numbers and passes the result to the callback
-
+const sumNums = (x, y, cb) => {
+  cb(x + y);
+};
 
 sumNums(5, 10, (sum) => {
   console.log(`The sum is ${sum}.`);
 });
 
 // Write a function called multiplyNums that adds two numbers and passes the result to the callback
+const multiplyNums = (x, y, cb) => {
+  cb(x * y);
+};
 
 multiplyNums(5, 10, (product) => {
   console.log(`The product is ${product}.`);
 });
 
 // Write a function called contains that checks if an item is present inside of the given array.
 // Pass true to the callback if it is, otherwise pass false
+const contains = (arr, item, cb) => {
+  if(arr.includes(item) === true) {
+    cb(item);
+  }
+  cb();
+}
 
 contains(foods, 'ribeye', (result) => {
   console.log(result ? 'ribeye is in the array' : 'ribeye is not in the array');
 });
 
 // Write a function called removeDuplicates that removes all duplicate values from the given array.
 // Pass the array to the callback function.  Do not mutate the original array.
-
+//I know i need map but i dont really get how map works
```

Comment:
That's fine, you can simply run a for(loop) and keep track of each element, if element == next element you can splice it out.. it's ok this is a difficult one.